stdlib/public/core/StringBridge.swift

```diff
@@ -195,6 +239,23 @@ internal func _cocoaStringCopyUTF8(
   _ target: _CocoaString,
   into bufPtr: UnsafeMutableBufferPointer<UInt8>
 ) -> Int? {
+  if let str = asFormerlyTaggedConstantString(source) {
+    let buf = UnsafeBufferPointer<UInt8>(
+          start: str.asciiContentsPointer,
+          len: str.utf16Length
+    )
+    var ptr = mbp.baseAddress._unsafelyUnwrappedUnchecked
+    var numWritten = 0
+    
+    for char in buf {
+      guard numWritten < bufPtr.count else { return nil }
+      // always ascii
+      ptr.initialize(to: char)
+      destination += 1
+      numWritten += 1
```

Comment:
How does this loop work? Are we just overwriting the first character repeatedly?